changelog_unreleased/cli/10274.md

```diff
@@ -0,0 +1,14 @@
+#### Add `--no-plugin-search` to turn off plugin autoloading (#10274 by @fisker)
+
+To turn off plugin autoloading, use `{ pluginSearchDirs: false }` in JavaScript code or `--no-plugin-search` when using Prettier CLI.
+
+If you don't use any plugins, you can also add `{ pluginSearchDirs: false }` to your config file.
```

Comment:
```suggestion
To turn off plugin autoloading, use `{ pluginSearchDirs: false }` in JavaScript code or `--no-plugin-search` when using Prettier CLI. You can also add `{ pluginSearchDirs: false }` to your config file.
```

Use case:

```js
// prettier.config.js

module.exports = {
  trailingComma: "all",

  plugins: [
    require("prettier-plugin-packagejson"),
    require("prettier-plugin-sh"),
  ],
  pluginSearchDirs: false
};
```

Adding `pluginSearchDirs` does not mean that I ‘don't use any plugins’.

---

Side thought: Perhaps, `no-plugin-search` mode can become default default behaviour in v3 because it’s incompatible with PnP. That’s something to discuss separately I guess.

Reply:
> no-plugin-search mode can become default default behaviour in v3 

I don't known, but I'm going to add into my config, and I'm planning to use it in our codebase including tests.